src/Symfony/Component/Serializer/Normalizer/ChainDenormalizer.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+use Symfony\Component\Serializer\Exception\PartialDenormalizationException;
+use Symfony\Component\Serializer\SerializerAwareInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * This holds a collection of denormalizers. It tries to be smart how it selects
+ * the correct Denormalizer.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class ChainDenormalizer implements ChainDenormalizerInterface, SerializerAwareInterface
+{
+    private const SCALAR_TYPES = [
+        'int' => true,
+        'bool' => true,
+        'float' => true,
+        'string' => true,
+    ];
+
+    /**
+     * @deprecated since Symfony 7.1
```

Comment:
I don't think it's needed as the property is private

Reply:
at least this allows not forget about removing the property when we'll start working on 8.0